### DISCUSSION TITLE
Proyecto Bidón

### DISCUSSION THREAD

**Discussion by NataliaAlvarezIspc** [NataliaAlvarezIspc/proyecto-ispc-ecommerce#59] [General]
Creo esta discusión para armar la redacción para la wiki justificando porque no continuamos con el proyecto del año pasado.

**Comment by NataliaAlvarezIspc** [comment]
Ayuda @rpgrca @claubott @Brian-Almada

**Comment by NataliaAlvarezIspc** [comment]
Creamos un proyecto desde cero, dado que el proyecto que realizamos el cuatrimestre pasado no cumplía con los requerimientos necesarios para lo solicitado en este trayecto. Consideramos que era inviable y había que realizar demasiadas modificaciones, pudiendo no alcanzar a cumplir con las etapas propuestas

**Comment by rpgrca** [comment]
Decidimos crear un proyecto desde cero dado que el proyecto anterior era de una naturaleza muy distinta (ya que una de las condiciones para la selección era que fuese un proyecto poco común, no simplemente "un carrito de compras"). En el sistema de subasta la venta no era posible si no había una puja, la puja no era posible si no había un lote y un lote no era posible si no existía un artículo, agregar un e-commerce haría que existan ventas que rompen con esa regla fundamental de nuestro sistema, que existan ventas sin pujas y por consiguiente hubiésemos tenido que agregar una segunda tabla de ventas o agregar código para validar si la venta venía de subasta o directa lo que, debido al escaso tiempo que vamos a tener para implementar la solución, hubiese complejizado el sistema dejándolo muy difícil de testear. Consideramos por lo tanto que era inviable.

**Reply by rpgrca** [reply]
Algo así con una explicación técnica podría ser.

**Reply by BraianTroncoso** [reply]
El proyecto anterior tenía un sistema de subasta en el que la venta dependía de una puja, que a su vez dependía de la existencia de un lote y un artículo. Para mantener la integridad del sistema, se decidió no agregar un e-commerce que permitiera ventas directas sin pujas, lo que habría requerido agregar una segunda tabla de ventas o código adicional para validar si la venta era de subasta o directa. Debido al limitado tiempo para implementar la solución y evitar complicaciones, se decidió que esta opción no era viable.
